tq/adapterbase.go

```diff
@@ -194,12 +194,14 @@ func (a *adapterBase) worker(workerNum int, ctx interface{}) {
 var httpRE = regexp.MustCompile(`\Ahttps?://`)
 
 func (a *adapterBase) newHTTPRequest(method string, rel *Action) (*http.Request, error) {
-	if !httpRE.MatchString(rel.Href) {
-		urlfragment := strings.SplitN(rel.Href, "?", 2)[0]
+	href := a.apiClient.Endpoints.NewEndpoint(a.direction.String(), rel.Href).Url
```

Comment:
I have some questions about how this is going to work with existing implementations.

For example, if I have the following, is this going to still work, or will things break?

```
[url "ssh://git@github.com/"]
    insteadOf = "https://github.com/"
```

Similar questions apply to how this will work with Bitbucket and GitLab's implementations. My gut tells me this will work fine for GitHub, but it will probably not for Bitbucket. If that's the case, we'll probably need to leave this until a 3.0 version where we have an SSH-based protocol, since otherwise this will break working configurations.